src/Migrations/AddXHPChildrenDeclarationMethodMigration.hack

```diff
@@ -0,0 +1,443 @@
+    return C\any(
+      $classes,
+      $class ==> !C\is_empty(
+        $class->getBody()
+          ->getElements()
+          ?->getChildrenOfType(XHPChildrenDeclaration::class) ??
+          vec[],
+      ),
+    );
+  }
+
+  private static function addUseNamespaceToScript(Script $in): Script {
+    $decls = $in->getDeclarations();
+    if (!self::scopeNeedsUseNamespace($decls)) {
+      return $in;
+    }
+
+    $before = C\firstx($decls->getChildrenOfType(IDeclaration::class));
+    $t = $before->getFirstTokenx();
+    $use = self::getUseNamespace();
+    if ($t->getLeading() !== $t->getLeadingWhitespace()) {
+      $ut = $use->getFirstTokenx();
+      return $in->withDeclarations(
+        $in->getDeclarations()->insertBefore(
+          $before,
+          $use->replace($ut, $ut->withLeading($t->getLeading())),
+        ),
+      )
+        ->replace($t, $t->withLeading(null));
+    }
+    return $in->withDeclarations(
+      $decls->insertBefore($before, self::getUseNamespace()),
+    );
+  }
+
+  private static function addUseNamespaceToNamespaceBlock(
+    NamespaceDeclaration $in,
+  ): NamespaceDeclaration {
+    // Can't use XHP in namespaces, so if we're not in the root namespace,
+    // we have nothing to do.
+    if ($in->hasName()) {
+      return $in;
+    }
+
+    // Don't have to worry about empty bodies as `namespace;` is invalid,
+    // and `namespace foo;` is caught above.
+
+    $body = $in->getBody() as NamespaceBody;
+    $decls = $body->getDeclarationsx();
+    if (!self::scopeNeedsUseNamespace($decls)) {
+      return $in;
+    }
+
+    $use = self::getUseNamespace();
+    $ws = $decls->getFirstTokenx()->getLeadingWhitespace();
+    if ($ws !== null) {
+      $t = $use->getFirstTokenx();
+      $use = $use->replace($t, $t->withLeading(new NodeList(vec[$ws])));
+    }
+
+    return $in->withBody(
+      $body->withDeclarations(
+        $decls->insertBefore(
+          C\firstx($decls->getChildrenOfType(IDeclaration::class)),
+          $use,
+        ),
+      ),
+    );
+  }
+
+  <<__Memoize>>
+  private static function getUseNamespace(): NamespaceUseDeclaration {
+    $s = new NodeList(vec[new WhiteSpace(' ')]);
+    return new NamespaceUseDeclaration(
+      new UseToken(null, $s),
+      new NamespaceToken(null, $s),
+      new NodeList(vec[new ListItem(
+        new NamespaceUseClause(
+          null,
+          new QualifiedName(
+            new NodeList(vec[
+              new ListItem(
+                new NameToken(null, null, 'Facebook'),
+                new BackslashToken(null, null),
+              ),
+              new ListItem(
+                new NameToken(null, null, 'XHP'),
+                new BackslashToken(null, null),
+              ),
+              new ListItem(new NameToken(null, $s, 'ChildValidation'), null),
+            ]),
+          ),
+          new AsToken(null, $s, 'as'),
+          new NameToken(null, null, 'XHPChild'),
```

Comment:
no - it would if we just did `use Foo`, but with `use namespace` it's unambigious - this is the same as how we have the `vec` builtin type and the `Vec` HSL namespace without conflcits

Reply:
Oh right `use *namespace*`.